### PR TITLE
Billing: ES6ify and modernize BillingHistory and BillingReceipt

### PR DESCRIPTION
**Pull request by tyxla**
This PR ES6ifies and modernizes the `BillingHistory` and `BillingReceipt` components and removes most of the ESLint warnings. There are no functional changes, only syntactic sugar and improvements.

This PR is part of the billing history reduxification effort (#10554).

To test:

* Checkout this branch
* Verify there are no changes or regressions in the Billing History page - `/me/purchases/billing`
* Verify there are no changes or regressions in the Billing Receipt page - `/me/purchases/billing/$transactionId`

### CODE REVIEW

File: client/me/billing-history/receipt.jsx [hunk 1]
```diff
@@ -75,69 +79,72 @@ module.exports = React.createClass( {
...
 		);
 	},
 
-	renderLineItems: function( transaction ) {
-		var items, creditBadge;
-
-		items = transaction.items.map( function( item ) {
+	renderLineItems() {
+		const { transaction, translate } = this.props;
+		const creditBadge = <span className="billing-history__credit-badge">{ translate( 'Refund' ) }</span>;
```
Comment (gwwar) [inline client/me/billing-history/receipt.jsx]: Maybe move this inline?
```
{ transaction.credit && <span className="billing-history__credit-badge">{ translate( 'Refund' ) }</span> }
```

File: client/me/billing-history/receipt.jsx [hunk 2]
```diff
@@ -75,69 +79,72 @@ module.exports = React.createClass( {
...
+		const { transaction, translate } = this.props;
+		if ( ! transaction.cc_name && ! transaction.cc_email ) {
 			return null;
 		}
 
 		return (
 			<li className="billing-history__billing-details">
-				<strong>{ this.translate( 'Billing Details' ) }</strong>
+				<strong>{ translate( 'Billing Details' ) }</strong>
 				<div contentEditable="true">{ transaction.cc_name }</div>
```
Comment (gwwar) [inline client/me/billing-history/receipt.jsx]: Out of curiosity why is there a contentEditable on these fields?

Comment (tyxla) [inline client/me/billing-history/receipt.jsx]: @gwwar these two fields are displayed only when that purchase has been made with a credit card. And they allow the user to change the name or email directly in the receipt before printing it. Here is a preview of how the editable fields look like when rolled over with the mouse: 



This appears to have been introduced in 6648-gh-calypso-pre-oss.

Comment (gwwar) [inline client/me/billing-history/receipt.jsx]: Oh printing 😄 I was wondering what use case that was for.

**Pull request merged**